File: tournaments/sumBelowBound/sumBelowBound.cpp
int sumBelowBound(int bound) {

  int left = 0,
      right = bound + 1;
  while (right - left > 1) {
    int middle = (right + left) / 2;
    if (middle * (middle + 1) / 2 <= bound) {
      left = middle;
    } else {
      right = middle;
    }
  }

  return left;
}

File: tournaments/sumBelowBound/sumBelowBound.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub sumBelowBound {
    my ($bound) = @_;

    my $n = 1;

    while ($n * ($n + 1) >> 1 <= $bound) {
        ++$n;
    }

    return $n - 1;
}
